pennylane_lightning/lightning_qubit/lightning_qubit.py

```diff
@@ -78,11 +78,23 @@ def simulate(circuit: QuantumScript, state: LightningStateVector, mcmc: dict = N
     state.reset_state()
     has_mcm = any(isinstance(op, MidMeasureMP) for op in circuit.operations)
     if circuit.shots and has_mcm:
-        mid_measurements = {}
-        final_state = state.get_final_state(circuit, mid_measurements=mid_measurements)
-        return LightningMeasurements(final_state, **mcmc).measure_final_state(
-            circuit, mid_measurements=mid_measurements
+        results = []
+        aux_circ = qml.tape.QuantumScript(
+            circuit.operations,
+            circuit.measurements,
+            shots=[1],
+            trainable_params=circuit.trainable_params,
         )
+        for _ in range(circuit.shots.total_shots):
+            state.reset_state()
```

Comment:
do we need to reset the state inside the for loop or this can be applied only once in the first iteration?
```suggestion
        )
        state.reset_state()
        for _ in range(circuit.shots.total_shots):
```

Reply:
Yes, this is because we want to run n-shots times the tape from scratch.